sktime/transformations/series/basisfunction.py

```diff
@@ -0,0 +1,209 @@
+"""Radial Basis Function (RBF) Transformer for Time Series Data."""
+
+__author__ = ["phoeenniixx"]
+from enum import Enum
+
+import numpy as np
+import pandas as pd
+
+from sktime.transformations.base import BaseTransformer
+
+
+class RBFType(Enum):
+    """
+    Enumeration for the types of Radial Basis Functions (RBFs) that can be applied.
+
+    Available RBF types:
+    - GAUSSIAN: Uses a Gaussian function.
+    - MULTIQUADRIC: Uses a multiquadric function.
+    - INVERSE_MULTIQUADRIC: Uses an inverse multiquadric function.
+    """
+
+    GAUSSIAN = "gaussian"
+    MULTIQUADRIC = "multiquadric"
+    INVERSE_MULTIQUADRIC = "inverse_multiquadric"
+
+
+class RBFTransformer(BaseTransformer):
+    """
+    A custom transformer that applies Radial Basis Functions (RBFs) to time series data.
+
+    This transformer allows the user to apply different RBFs such as Gaussian,
+    multiquadric, and inverse multiquadric to time series data. It can be used
+    as a feature transformation technique to augment or modify the input data
+    in a machine learning pipeline.
+
+    RBFs are functions that depend on the distance between a given point and
+    a center point. The transformation involves applying a mathematical
+    function (such as Gaussian or multiquadric) to this distance. Each
+    center has an influence over the surrounding data points, and the
+    influence decreases as the distance from the center increases. The
+    `gamma` parameter controls the "spread" of the RBF,
+    influencing how far the influence of each center reaches.
+
+    The Gaussian RBF is calculated as `exp(-gamma * (x - c)^2)`, where `x` is
+    the input data point and `c` is the center. Multiquadric and inverse multiquadric
+    functions are calculated with different equations, providing different
+    shapes for the transformation.
+
+    This implementation is inspired by the `RepeatingBasisFunction` transformer
+    from the `scikit-lego` package:
+    https://github.com/koaning/scikit-lego/blob/main/sklego/preprocessing/repeatingbasis.py
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:transform-labels": "None",
+        "scitype:instancewise": True,
+        "X_inner_mtype": ["pd.DataFrame", "pd.Series", "np.ndarray"],
+        "y_inner_mtype": "None",
+        "univariate-only": False,
+        "requires_y": False,
+        "fit_is_empty": False,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "authors": ["phoeenniixx"],
+    }
+
+    def __init__(self, centers=None, gamma=1.0, rbf_type=RBFType.GAUSSIAN):
+        self.centers = centers
+        self.gamma = gamma
+        self.rbf_type = rbf_type
```

Comment:
I think this should simply be string, otherwise the user will have to (unnecessarily) import the enum.